### PR TITLE
kvui: allow sorting hints in the hint tab

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
allow sorting hints in the hint tab

## How was this tested?
SortHintTab

## If this makes graphical changes, please attach screenshots.
SortHintTab

### CODE REVIEW

**Comment by beauxq** [comment]
I played through a game with this. Didn't see any problems.

I noticed that alphabetic sorting is uppercase before lowercase - maybe not what some people would expect, but not a problem. I know that's what's easiest.

**Comment by NewSoupVi** [comment]
> I noticed that alphabetic sorting is uppercase before lowercase - maybe not what some people would expect (and different from the web site tracker), but not a problem. I know that's what's easiest.

I personally would **absolutely** say that it should be changed if possible

File: kvui.py [hunk 1]
```diff
@@ -324,24 +324,44 @@ def on_touch_down(self, touch):
...
                     return self.parent.select_with_touch(self.index, touch)
+        else:
+            parent = self.parent
+            parent.clear_selection()
+            parent: HintLog = parent.parent
+            # find correct column
+            for child in self.children:
+                if child.collide_point(*touch.pos):
+                    key = child.sort_key
+                    parent.hint_sorter = lambda element: remove_between_brackets.sub("", element[key]["text"])
```
Comment (Berserker66) [inline kvui.py:349]:
```suggestion
                    parent.hint_sorter = lambda element: remove_between_brackets.sub("", element[key]["text"]).lower()
```

**Review by qwint — APPROVED** [review]
Don't fully follow the kivy code, but light testing on the branch with a fully hinted world seems to sort fine

I would want the ability for a sort of 'sub sort' where it remembers the last sort you did and keeps it together for all entries that are equivalent under the new sorting method
ex. sort by location name, then sort by item name, all items named 'coin' are put after the 'apple' item, but are sorted by their location name inside the 'coin' subset

BUT that would be a nice to have and don't want it to hold this functionality up if it doesn't have to.

**Review by beauxq — APPROVED** [review]
kivy is difficult to get good typing with, so the fact the I don't see problems with the code is not as valuable as it usually is...

(This is my biggest problem with kivy, but as has been discussed elsewhere, I don't have a good alternative.)

I tested it about a month ago and it worked. And I looked at the commit that addressed what I talked about.

LGTM

**Review by Ixrec — APPROVED** [review]
Confirmed this works as expected on my current async game.

image

The sort order seems correct for every column and direction I try. In some cases the text gets clipped on the top and bottom (this is also visible in Berserker's gif), but I assume that's preexisting kivy behavior and not a concern with this specific PR.

The code also makes sense to me. Redefining a static method instead of, say, a static string feels weird to me, but maybe that's just my lack of Python experience.

**Pull request merged**